vignettes/sir_models.Rmd

```diff
@@ -0,0 +1,680 @@
+                           step = 1,
+                           n_particles = n_particles,
+                           n_threads = 4L,
+                           seed = 1L)
+sir_model_2$state()
+```
+
+Let's run this epidemic forward, and plot the trajectories:
+
+```{r fig.height=5, fig.width=7}
+n_steps <- 200
+x <- array(NA, dim = c(sir_model_2$info()$len, n_particles, n_steps))
+
+for (t in seq_len(n_steps)) {
+  x[ , , t] <- sir_model_2$run(t)
+}
+time <- x[1, 1, ]
+x <- x[-1, , ]
+
+par(mar = c(4.1, 5.1, 0.5, 0.5), las = 1)
+matplot(time, t(x[1, , ]), type = "l",
+         xlab = "Time", ylab = "Number of individuals",
+         col = "#8c8cd9", lty = 1, ylim = range(x))
+matlines(time, t(x[2, , ]), col = "#cc0044", lty = 1)
+matlines(time, t(x[3, , ]), col = "#999966", lty = 1)
+legend("left", lwd = 1, col = c("#8c8cd9", "#cc0044", "#999966"), legend = c("S", "I", "R"), bty = "n")
+```
+
+Many other methods are available on the dust object to support inference, but typically you will use the `mcstate` package instead of calling these directly. However, if you wish to simulate state space models with set parameters, this can be done entirely using the commands above from `dust`.
+
+# Inferring parameters with [`mcstate`](https://mrc-ide.github.io/odin.dust/)
+
+Typically these models will be used in conjunction with observed data to infer the most likely parameter values, most likely trajectories up to the present (nowcast), and make forecasts using these parameters. The `mcstate` package implements particle filter and pMCMC methods to make all of this this possible from `dust` models.
+
+A comparison function is needed to define a likelihood, i.e. the probability of observing this trajectory, given the data. This is combined with a prior to generate a posterior distribution for each parameter, which in the SIR model gives the distribution of likely values for $\beta$ and $\gamma$. 
+
+This is straightforward for a deterministic model, but for a stochastic model looking at a single run can easily bias these estimates. `mcstate` therefore implements a particle filter, also known as sequential Monte Carlo (SMC), which runs multiple trajectories, and at each stage where data is available resamples them in proportion to their likelihood. This produces an unbiased estimate of the likelihood for a given set of parameters.
+
+The anatomy of an mcstate particle filter consists of three components:
+
+- A set of observations to fit the model to, generated using `particle_filter_data()`.
+- A model to fit, which must be a dust generator.
+- A comparison function, which is an R function which calculates the likelihood of the state given the data.
+
+## Model data
+
+To demonstrate this, we will attempt to infer the parameters $\beta$ and $\gamma$ using daily case counts. In reality these counts would be measured, but here we generate them from a single run of the model, and adding some constant noise proportional to the epidemic size:
+
+```{r}
```

Comment:
My suggestion here is that you provide the data as a fait accompli. You can tell the reader that they were generated from the model (or not), but read them from a csv or hide the sausage factory here

Reply:
Excellent idea, this is confusing